core/backend/src/SqliteStatement.ts

```diff
@@ -75,13 +75,19 @@ export class SqliteStatement implements IterableIterator<any>, IDisposable {
    * or not. See [SQLite docs](https://www.sqlite.org/c3ref/stmt_readonly.html) for details.
    */
   public get isReadonly(): boolean {
-    return this._stmt!.isReadonly();
+    if (undefined === this._stmt) {
```

Comment:
@khanaffan `assert` here and elsewhere on `this._stmt`?

Reply:
or do nothing. If you attempt to use a statement before you prepare it you will crash (that's why `isPrepared` exists)